bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/CallOptionsFactory.java

```diff
@@ -93,24 +93,24 @@ public ConfiguredCallOptionsFactory(CallOptionsConfig config) {
       if (contextDeadline != null) {
         return CallOptions.DEFAULT.withDeadline(contextDeadline);
       } else if (config.isUseTimeout() && request != null) {
-        int timeout =
-            isLongRequest(request) ? config.getLongRpcTimeoutMs() : config.getShortRpcTimeoutMs();
+        int timeout = getRequestTimeout(request);
         return CallOptions.DEFAULT.withDeadline(Deadline.after(timeout, TimeUnit.MILLISECONDS));
       } else {
         return CallOptions.DEFAULT;
       }
     }
 
     /**
-     * @param request
-     * @return true if this is a {@link MutateRowsRequest} or a {@link ReadRowsRequest} that's a
-     *     scan.
+     * @param request an RPC request.
+     * @return timeout in milliseconds as per the Request type.
      */
-    public static boolean isLongRequest(Object request) {
```

Comment:
Deleting this would be a breaking change - can we keep it or deprecate it?

Reply:
Thanks for pointing it out. I have reverted this method and added a deprecation message.